piet-direct2d/src/lib.rs

```diff
@@ -456,21 +457,43 @@ impl<'a> RenderContext for D2DRenderContext<'a> {
     fn capture_image_area(&mut self, rect: impl Into<Rect>) -> Result<Self::Image, Error> {
         let r = rect.into();
 
+        let (dpi_scale, _) = self.rt.get_dpi_scale();
+        let dpi_scale = dpi_scale as f64;
+
+        let transform_matrix = self.rt.get_transform();
+        let affine_transform = matrix3x2f_to_affine(transform_matrix);
+
+        let device_size = Point {
+            x: r.width() * dpi_scale,
+            y: r.height() * dpi_scale,
+        };
+        // TODO: This transformation is untested with the current test pictures
+        let device_size = affine_transform * device_size;
+
+        let device_origin = Point {
+            x: r.x0 * dpi_scale,
+            y: r.y0 * dpi_scale,
+        };
+        // TODO: This transformation is untested with the current test pictures
+        let device_origin = affine_transform * device_origin;
+
         let mut target_bitmap = self.rt.create_blank_bitmap(
-            r.width() as usize,
-            r.height() as usize,
+            device_size.x as usize,
+            device_size.y as usize,
             D2D1_ALPHA_MODE_PREMULTIPLIED,
         )?;
 
-        let dest_point = to_point2u((0.0f32, 0.0f32));
-        let src_rect = rect_to_rectu(Rect {
-            x0: r.x0,
-            y0: r.y0,
-            x1: r.width(),
-            y1: r.height(),
-        });
+        let src_rect = Rect {
+            x0: device_origin.x,
+            y0: device_origin.y,
+            x1: device_size.x * dpi_scale,
+            y1: device_size.y * dpi_scale,
+        };
```

Comment:
This `src_rect` code right here is wrong and causing the werid Direct2D results. (It's the good old magic multipleir part, so not a big surprise to anyone.) The simplest change would be to do `x1: device_origin.x + device_size.x` instead of `device_size.x * dpi_scale`. However now that `device_size` is an actual `Size`, there is an even better way:

```rust
let src_rect = Rect::from_origin_size(device_origin, device_size);
```